### PR TITLE
Deno CI in Travis

### PR DESCRIPTION
**Pull request by tcarrio**
## Summary

As suggested in issue #43 , it would be nice to have some kind of automation around the Deno runtime remaining compatible with this library. I have some experience with CI tooling and figured I could help extend the current Travis config.

## Details

- Providing build matrix to support multilanguage builds in Travis
- Since no official Docker images exist for Deno yet, using the popular images provided by github.com/hayd/deno-docker
- Custom Dockerfile providing deno runtime with the full project
- New pipeline commands using native deno functionality which:
  - run src/index.ts
  - bundle src/index.ts
  - lint src/index.ts
- Note: No test commands as Deno has its own test framework that would have to replace the existing Jest tests

This provides minimal updates around linting errors (implemented one of the type checking functions using TypeScript type guard syntax and disabled an `any` linting error) as well.

### CODE REVIEW

**Comment by tcarrio** [comment]
I updated it to follow a similar design to your current Node.js commands and also remove containers after the run completes by default.

**Comment by tcarrio** [comment]
Caught one last time by trailing whitespace. The touch-ups should be complete now- sorry for the noise.

File: .travis.yml [hunk 1]
```diff
@@ -1,12 +1,23 @@
...
+matrix:
+  include:
+    - language: node_js
+      node_js:
+        - lts/*
+        - 14
+        - 13
+        - 12
+        - 11
+        - 10
```
Comment (balazsbotond) [inline .travis.yml]: Looks like Travis takes the first item in this array and ignores the rest, so it only runs the build for `lts/*`.

See the generated JSON build config from [here](https://travis-ci.com/github/balazsbotond/urlcat/builds/187936230/config):

```json
{
  "language": "ruby",
  "os": [
    "linux"
  ],
  "dist": "xenial",
  "jobs": {
    "include": [
      {
        "language": "node_js",
        "node_js": "lts/*",
        "script": [
          "npm run lint",
          "npm test",
          "npm run coverage"
        ]
      },
      {
        "language": "shell",
        "services": [
          "docker"
        ],
        "script": [
          "docker build -f Dockerfile.deno -t urlcat-deno .",
          "docker run --rm urlcat-deno run ./src/index.ts",
          "docker run --rm urlcat-deno bundle ./src/index.ts",
          "docker run --rm urlcat-deno lint --unstable src/",
          "docker rmi urlcat-deno"
        ]
      }
    ]
  }
}
```

I think the `language: node_js` section needs to be repeated for each node version.

**Review by balazsbotond — CHANGES_REQUESTED** [review]
Thank you! This seems to work great - except the node build only runs on one version. I added a comment explaining what I think needs to be changed but feel free to suggest another solution.

**Comment by tcarrio** [comment]
This was rebased on master. There appears to be an import from `qs` now, I wonder if that will break our Deno compatibility here.

**Comment by tcarrio** [comment]
As somewhat expected, external libraries breaks the Deno compatibility without some considerable effort. It appears the latest merge from #51 to resolve #6.

**Comment by balazsbotond** [comment]
Sorry about that; I didn't realize it would break Deno compatibility. I've created a new branch `2.x` from the commit before #51 was merged - could you please retarget your PR on it so I can merge it? After that, I will deal with the qs dependency. Sorry about the inconvenience.

**Comment by tcarrio** [comment]
Thanks for branching that out! I'm rebasing the work on the new 2.x branch. Should be up shortly :+1:

**Comment by tcarrio** [comment]
I opened #57 which is based on the 2.x branch

**Comment by balazsbotond** [comment]
Thank you! Closing this one because I've just merged #57

**Pull request closed without merging**